docs/docs/neptune_for_amundsen.md

```diff
@@ -0,0 +1,152 @@
+for the Neptune cluster to be able to access the files, there must be a S3 VPC endpoint in the VPC
+in which the Neptune cluster runs.  (It feels like there's a long story behind this.)  Further,
+the Neptune cluster needs an assumable that allows it to access the files upon which the Bulk
+Loader is invoked.  (This is potentially very simple... but also potentially extremely complicated
+because the policies of the S3 VPC Endpoint, the Bucket permissions on the S3 Bucket used, the
+per-Object permissions on the files in the Bucket, as well as those of the assumed Role.)
+
+The terraform example creates: the S3 VPC Endpoint, a role for Neptune which gives it readonly
+access to all S3 buckets and object (in that account), and a bucket.
+
+### Create Neptune cluster and instances
+
+Possibly, you will create a Neptune Subnet Group.  If you're using the default VPC, a Neptune
+Subnet Group already exists.  Otherwise, you create one. 
+
+Then Create a cluster and (at least) one instance, using the Security Group and assumable Role.
+The cluster's `cluster_identifier` and the instance's `identifier` will become part of the DNS
+names so choose them appropriately.  The `instance_class` configures the amount of RAM, CPU, and
+IO available to the instance, as well is being the primary factor in cost.  (Chosing this is
+experimental, and beyond the scope of this guide.)  
+
+There are other interesting paramters related to admininstration: the maintenance window,
+automatic minor version upgrades, deletion protection, backup retentoin, encryption-at-rest and
+key management for that, default query execution timeouts.  (These are also beyond the scope of
+this.)
+
+
+The terraform example configures a Neptune Subnet Group, a Neptune Cluster, and 2 Neptune Instances.
+
+
+### Create user(s) and give it(them) access to Neptune and S3
+
+The metadata and dataloader processes require access to Neptune and S3.  In the most direct setup,
+each would use a distinct IAM user and receive just those permissions required to access Neptune
+and S3.  An another, you might run the metadata service and dataloader processes in EC2 and you
+would grant whatever Role they assume the required permissions for Neptune and S3.
+
+The terraform example creates a Policy which gives the metadata and dataloader services sufficient
+access, and creates 2 Users and gives attaches the Policy.
+
+
+### Create public endpoint for Neptune
+
+Neptune does not have public endpoints.  The state of the art seems to be using
+a [loadbalancer](
+https://github.com/aws-samples/aws-dbs-refarch-graph/tree/master/src/connecting-using-a-load-balancer)
+as well as glue to allow the mismatch in hostnames.
+
+
+The terraform sets up a publicly available NLB -- albeit statically configured, so is only suitable
+for development or other deployments where one could tolerate downtime as instances are
+reconfigured.  (Or you could apply the CloudWatch + Lambda approach and minimize that downtime.)
```

Comment:
I don't think the "CloudWatch + Lambda" approach is described elsewhere in this doc, seems out of scope and maybe not worth mentioning here. WDYT?

```suggestion
reconfigured. 
```

Reply:
I think it's possibly useful as a breadcrumb or exercise for the reader if one cannot tolerate such downtime.

Reply:
Sounds good to me. I noticed after the fact that some comments in the terraform code do dig into this a little bit more so it seems reasonable to leave 👍